### PR TITLE
8297697: RISC-V: Add support for SATP mode detection

### PR DESCRIPTION
**Pull request by zifeihan**
Hi,
Please review this backport to riscv-port-jdk11u.
Backport of [JDK-8297697](https://bugs.openjdk.org/browse/JDK-8297697), and [JDK-8301067](https://bugs.openjdk.org/browse/JDK-8301067). JDK-8301067 only modifies the log description information introduced by JDK-8297697.

After this patch, If it is an unsupported satp.mode , the message will be as follows:
```
root@qemuriscv64:~/jdk/bin# ./java -version
Error occurred during initialization of VM
Unsupported satp mode: sv57

root@qemuriscv64:~/jdk/bin# cat /proc/cpuinfo
processor       : 0
hart            : 0
isa             : rv64imafdch_zicbom_zicboz_zicntr_zicsr_zifencei_zihintpause_zihpm_zba_zbb_zbs_sstc
mmu             : sv57
mvendorid       : 0x0
marchid         : 0x0
mimpid          : 0x0
```

Testing:

- [ ] Run tier1 tests on SOPHON SG2042 (release)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8297697](https://bugs.openjdk.org/browse/JDK-8297697): RISC-V: Add support for SATP mode detection (**Enhancement** - P4)
 * [JDK-8301067](https://bugs.openjdk.org/browse/JDK-8301067): RISC-V: better error message when reporting unsupported satp modes (**Enhancement** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**) ⚠️ Review applies to [856643f1](https://git.openjdk.org/riscv-port-jdk11u/pull/19/files/856643f195e0c90ffdb28405c8fb690ff012647f)
 * [Feilong Jiang](https://openjdk.org/census#fjiang) (@feilongjiang - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/riscv-port-jdk11u.git pull/19/head:pull/19` \
`$ git checkout pull/19`

Update a local copy of the PR: \
`$ git checkout pull/19` \
`$ git pull https://git.openjdk.org/riscv-port-jdk11u.git pull/19/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19`

View PR using the GUI difftool: \
`$ git pr show -t 19`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/riscv-port-jdk11u/pull/19.diff">https://git.openjdk.org/riscv-port-jdk11u/pull/19.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/riscv-port-jdk11u/pull/19#issuecomment-2043278637)

### CODE REVIEW

**Review by RealFYang — APPROVED** [review]

File: src/hotspot/cpu/riscv/vm_version_riscv.cpp [hunk 1]
```diff
@@ -33,11 +33,17 @@
...
 const char* VM_Version::_uarch = "";
+const char* VM_Version::_vm_mode = "";
 uint32_t VM_Version::_initial_vector_length = 0;
 
 void VM_Version::initialize() {
   get_os_cpu_info();
 
+  // check if satp.mode is supported, currently supports up to SV48(RV64)
+  if (get_satp_mode() > VM_SV48) {
+    vm_exit_during_initialization(err_msg("Unsupported satp mode: %s", _vm_mode));
```
Comment (feilongjiang) [inline src/hotspot/cpu/riscv/vm_version_riscv.cpp:44]: Looks like [JDK-8301067](https://bugs.openjdk.org/browse/JDK-8301067) was missed?

Comment (RealFYang) [reply src/hotspot/cpu/riscv/vm_version_riscv.cpp:44]: Good catch!

Comment (zifeihan) [reply src/hotspot/cpu/riscv/vm_version_riscv.cpp]: > Looks like [JDK-8301067](https://bugs.openjdk.org/browse/JDK-8301067) was missed?

Thanks, I've added a commit to fix it.

**Review by feilongjiang — APPROVED** [review]
looks good, thanks.

**Comment by zifeihan** [comment]
@RealFYang @feilongjiang : Thanks for the review.
/integrate

**Pull request closed without merging**